src/main/java/io/github/thebusybiscuit/slimefun4/core/config/SlimefunConfigManager.java

```diff
@@ -0,0 +1,275 @@
+            return null;
+        }
+    }
+
+    /**
+     * This method (re)loads all relevant config values into our cache.
+     * <p>
+     * <strong>Note that this method is not guaranteed to reload all settings.</strong>
+     * 
+     * @return Whether the reloading was successful and completed without any errors
+     */
+    public boolean reload() {
+        boolean isSuccessful = true;
+
+        try {
+            pluginConfig.reload();
+            itemsConfig.reload();
+            researchesConfig.reload();
+
+            researchesConfig.setDefaultValue("enable-researching", true);
+
+            enableBackwardsCompatibility = pluginConfig.getBoolean("options.backwards-compatibility");
+            enableResearching = researchesConfig.getBoolean("enable-researching");
+            enableFreeCreativeResearches = pluginConfig.getBoolean("researches.free-in-creative-mode");
+            enableResearchFireworks = pluginConfig.getBoolean("researches.enable-fireworks");
+            enableDuplicateBlockLogging = pluginConfig.getBoolean("options.log-duplicate-block-entries");
+            enableVanillaRecipesInGuide = pluginConfig.getBoolean("guide.show-vanilla-recipes");
+            enableGuideOnJoin = pluginConfig.getBoolean("guide.receive-on-first-join");
+            enableUpdater = pluginConfig.getBoolean("options.auto-update");
+            enableActionbarTalismanMessages = pluginConfig.getBoolean("talismans.use-actionbar");
+            enableCommandItemDropExcess = pluginConfig.getBoolean("options.drop-excess-sf-give-items");
+        } catch (Exception x) {
+            plugin.getLogger().log(Level.SEVERE, x, () -> "An Exception was caught while (re)loading the config files for Slimefun v" + plugin.getDescription().getVersion());
+            isSuccessful = false;
+        }
+
+        // Reload Research costs
+        for (Research research : SlimefunPlugin.getRegistry().getResearches()) {
+            try {
+                NamespacedKey key = research.getKey();
+                int cost = researchesConfig.getInt(key.getNamespace() + '.' + key.getKey() + ".cost");
+                research.setCost(cost);
+
+                if (!researchesConfig.getBoolean(key.getNamespace() + '.' + key.getKey() + ".enabled")) {
+                    // Disabling a research requires a restart (for now)
+                    isRestartRequired = true;
+                }
+            } catch (Exception x) {
+                plugin.getLogger().log(Level.SEVERE, x, () -> "Something went wrong while trying to update the cost of a research: " + research);
+                isSuccessful = false;
+            }
+        }
+
+        for (SlimefunItem item : SlimefunPlugin.getRegistry().getAllSlimefunItems()) {
+            if (!itemsConfig.getBoolean(item.getId() + ".enabled")) {
+                // Disabling an item requires a restart (for now)
+                isRestartRequired = true;
+            }
```

Comment:
Won't this always make it so a restart is required even if the something is disabled but the item config was not changed at all? You'd need to check the current state and see if it's changed (or am I being dumb?)

The if is a bit weird to read but it will check that the item is enabled and the new value is disabled
```suggestion
            if (!item.isDisabled() && !itemsConfig.getBoolean(item.getId() + ".enabled")) {
                // Disabling an item requires a restart (for now)
                isRestartRequired = true;
            }
```

Reply:
You are right. However I think semantically this would be better:

```suggestion
            if (item.isDisabled() != !itemsConfig.getBoolean(item.getId() + ".enabled")) {
                // Enabling/Disabling an item requires a restart (for now)
                isRestartRequired = true;
            }
```

Reply:
Yeah agreed